test/e2e/vmagent_test.go

```diff
@@ -0,0 +1,307 @@
+			tempVm = getVMInfo(t, data, vm.nodeName)
+		}
+		assert.Exactly(t, vm.ifName, tempVm.ifName, "Failed to verify uplink interface, Expected %s, Got %s", vm.ifName, tempVm.ifName)
+		assert.Exactly(t, vm.ip, tempVm.ip, "Failed to verify uplink IP, Expected %s, Got %s", vm.ip, tempVm.ip)
+	}
+	for _, vm := range vmList {
+		hash := sha1.New() // #nosec G401: not used for security purposes
+		io.WriteString(hash, vm.ifName)
+		hashedIfName := hex.EncodeToString(hash.Sum(nil))
+		// Generate ExternalEntity name from ExternalNode
+		eeName := vm.nodeName + "-" + hashedIfName[:interfaceNameLength]
+		t.Logf("Running verifyExternalEntityExistence")
+		verifyExternalEntityExistence(t, data, eeName, vm.nodeName, true)
+		t.Logf("Running verifyExternalNodeRealization")
+		verifyExternalNodeRealization(vm)
+	}
+}
+
+func getVMInfo(t *testing.T, data *TestData, nodeName string) (info vmInfo) {
+	var err error
+	var vm vmInfo
+	vm.nodeName = nodeName
+	var cmd string
+	cmd = "ip -o -4 route show to default | awk '{print $5}'"
+	vm.osType = "Linux"
+	_, vm.ifName, _, err = data.RunCommandOnNode(nodeName, cmd)
+	require.Nil(t, err, "Failed to run command %s on VM %s, err %v", cmd, nodeName, err)
+
+	vm.ifName = strings.TrimSpace(vm.ifName)
+	cmd = fmt.Sprintf("ifconfig %s | awk '/inet / {print $2}'| sed 's/addr://'", vm.ifName)
+	_, vm.ip, _, err = data.RunCommandOnNode(nodeName, cmd)
+	require.Nil(t, err, "Failed to run command %s on VM %s, err %v", cmd, nodeName, err)
+
+	vm.ip = strings.TrimSpace(vm.ip)
+	return vm
+}
+
+func getWindowsVMInfo(t *testing.T, data *TestData, nodeName string) (vm vmInfo) {
+	var err error
+	vm.nodeName = nodeName
+	vm.osType = "Windows"
+	cmd := fmt.Sprintf("powershell 'Get-WmiObject -Class Win32_IP4RouteTable | Where { $_.destination -eq \"0.0.0.0\" -and $_.mask -eq \"0.0.0.0\"} | Sort-Object metric1 | select interfaceindex | ft -HideTableHeaders'")
+	_, ifIndex, _, err := data.RunCommandOnNode(nodeName, cmd)
+	require.Nil(t, err, "Failed to run command %s on VM %s, err %v", cmd, nodeName, err)
+
+	vm.ifIndex = strings.TrimSpace(ifIndex)
+	cmd = fmt.Sprintf("powershell 'Get-NetAdapter -IfIndex %s | select name | ft -HideTableHeaders'", vm.ifIndex)
+	_, ifName, _, err := data.RunCommandOnNode(nodeName, cmd)
+	require.Nil(t, err, "Failed to run command %s on VM %s, err %v", cmd, nodeName, err)
+
+	vm.ifName = strings.TrimSpace(ifName)
+	cmd = fmt.Sprintf("powershell 'Get-NetIPAddress -AddressFamily IPv4 -ifIndex %s| select IPAddress| ft -HideTableHeaders'", vm.ifIndex)
+	_, ifIP, _, err := data.RunCommandOnNode(nodeName, cmd)
+	require.Nil(t, err, "Failed to run command %s on VM %s, err %v", cmd, nodeName, err)
+
+	vm.ip = strings.TrimSpace(ifIP)
+	return vm
+
+}
+
+func startAntreaAgent(t *testing.T, data *TestData, vm vmInfo) {
+	t.Logf("Starting antrea-agent on VM: %s", vm.nodeName)
+	var cmd string
+	if vm.osType == "Windows" {
+		cmd = "nssm start antrea-agent"
+	} else {
+		cmd = "sudo systemctl start antrea-agent"
+	}
```

Comment:
wouldn't it be easier to run it as a process for the duration of the test?

Reply:
From a user point of view, running it as a service is preferred. So that is the reason, we decided to go the service way for the test as well.